include/openmc/hdf5_interface.h

```diff
@@ -247,11 +247,11 @@ inline void read_dataset(
 {
   // Create buffer to read data into
   auto n = dataset_typesize(obj_id, name);
-  char* buffer = new char[n];
+  std::vector<std::string::value_type> buffer(n, '\0');
 
   // Read attribute and set string
-  read_string(obj_id, name, n, buffer, indep);
-  str = std::string {buffer, n};
+  read_string(obj_id, name, n, buffer.data(), indep);
+  str = std::string { buffer.begin(), buffer.end() };
```

Comment:
The `buffer` is never deleted. And I try to avoid `new` as much as possible. Unfortunately `std::string::data()` is `const char *` thus I used a vector. Two alternative solutions would be just to do the `delete` or to use a std::unique_ptr`